source/synthDrivers/sapi5.py

```diff
@@ -53,41 +55,116 @@ class SpeechVoiceEvents(IntEnum):
 	Bookmark = 16
 
 
-class SapiSink(object):
-	"""Handles SAPI event notifications.
-	See https://msdn.microsoft.com/en-us/library/ms723587(v=vs.85).aspx
+class SynthDriverAudioStream(COMObject):
 	"""
+	Implements IStream to receive streamed-in audio data.
+	Should be wrapped in an SpCustomStream
+	(which also provides the wave format information),
+	then set as the AudioOutputStream.
+	"""
+
+	_com_interfaces_ = [IStream]
 
 	def __init__(self, synthRef: weakref.ReferenceType):
 		self.synthRef = synthRef
+		self._writtenBytes = 0
 
-	def StartStream(self, streamNum, pos):
+	def ISequentialStream_RemoteWrite(self, this, pv, cb, pcbWritten):
```

Comment:
This is using a low level comtypes implementation. Have you considered the [high level approach](https://pythonhosted.org/comtypes/server.html#id9) instead?